### PR TITLE
Ensure Pod Share Stays Below Resource Limits

### PR DESCRIPTION
**Pull request by jignyasamishra**
Fix: #3255

**Description:**
This update resolves an issue related to Pod Share calculation within the resource limits. The prior implementation allowed the Pod Share to exceed the available resource capacity, potentially leading to resource overutilization. The modified logic now ensures that the Pod Share is accurately computed based on the maximum resource utilization found among CPU, memory, and init containers. If this calculated share exceeds the available resources, the code clamps it to a maximum value of one (1) to maintain stability and prevent overconsumption of resources. This enhancement guarantees that Pod Share remains within the confines of the available resources, mitigating any potential resource allocation inconsistencies or overusage scenarios.

**Changes Made:**
Adjusted Pod Share calculation logic to consider maximum resource utilization.
Implemented checks to ensure the computed Pod Share remains within available resource limits.
Added validation to limit the Pod Share value to one (1) when necessary, preventing overutilization of resources.

### CODE REVIEW

**Comment by Monokaix** [comment]
Welcome and thanks for your contribution！Will check this soon: )

File: pkg/scheduler/api/pod_info.go [hunk 1]
```diff
@@ -61,15 +61,36 @@ import (
...
+
+	// Adjust the share based on CPU, Memory, and maximum init container utilization
+	maxUtilization := math.Max(cpuSum, memorySum)
+	maxUtilization = math.Max(maxUtilization, initContainerMax)
+	if maxUtilization > 0 {
+		podShare := maxUtilization / 1000 // Convert to fractional value
+		if podShare > 1 {
+			podShare = 1
+		}
+		result.AddScalar(v1.ResourcePods, podShare)
```
Comment (Monokaix) [inline pkg/scheduler/api/pod_info.go:87]: Aha, seems it's misunderstanding, we assume the value 1 as scaler resource, which represents pod itself occupied a pod resource and this is used in preemption scenario, here the value should always be 1, and we should fix it in proportion plugin instead: )

Comment (jignyasamishra) [reply pkg/scheduler/api/pod_info.go:87]: Thank you much for your review!!!! I am working on it. I thought we need to adjust the resources accordingly. Can you provide me some more hints on the issue.

**Pull request closed without merging**